x/ccv/consumer/types/genesis.go

```diff
@@ -88,6 +90,9 @@ func (gs GenesisState) Validate() error {
 		if len(gs.MaturingPackets) != 0 {
 			return sdkerrors.Wrap(ccv.ErrInvalidGenesis, "maturing packets must be empty for new chain")
 		}
+		if gs.LastTransmissionBlockHeight.Height != 0 {
```

Comment:
Leaving comment here because github doesn't let you leave comments outside of diffs lol:

This entire function feels weird to me. Why are there so many checks on a new chain genesis state, and so few the restart? What about a restart where the handshake didn't complete? Isn't that basically like a new chain restart?

Reply:
Most of theses checks are validating that a "new chain" has all the required IBC sates. When a chain restarts, even when the handshake isn't completed, these states are handled by the IBC module.